webapp/src/ui/courses.js

```diff
@@ -1,100 +1,93 @@
-    filteredCourses = filteredCourses.filter(course => {
-      let entry = Object.values(course);
-      let entryString = entry.slice(0, entry.length)
-      let modifiedEntryString = entryString.join().toLowerCase();
+    const searchFilter = debouncedQuery.split(' ');
+
+    setFilteredCourses(courses.results.filter(course => {
+      const entry = Object.values(course);
+      const entryString = entry.slice(0, entry.length);
+      const modifiedEntryString = entryString.join().toLowerCase();
       return searchFilter.every(key => {
-        return modifiedEntryString.includes(key.toLowerCase())
+        return modifiedEntryString.includes(key.toLowerCase());
       });
-    });
-    return filteredCourses;
-  }
+    }));
+  }, [debouncedQuery, courses]);
 
-  handleChange = (event: object) => {
-    let searchKey = event.target.value.toLowerCase();
-    this.setState({ query: searchKey });
-    let data = this.filterCourses(searchKey);
-    this.setState({ filtered: data })
-  }
+  const onQueryChange = event => {
+    setQuery(event.target.value);
 
+    setIsFiltering(true);
+  };
 
-  render() {
-    const courses = this.props.courses;
-    const state = this.state;
-    const filtered = state.filtered;
-    const totalData = courses.total;
-    let data = courses.results;
-    if(filtered.length > 0 && filtered.length < data.length) {
-      data = filtered;
-    }
-    return (
-      <div className="courses">
-        <Span>
-          Matched {filtered.length} results
-        </Span>
-        <Span> Got {totalData ? totalData : 0} results </Span>
-        <Input
-          value={state.query}
-          onChange={(e) => this.handleChange(e)}
-          fullWidth={true}
-          placeholder="Search..."
-          autoFocus={true}
-        />
-        {!this.props.courses.loading
-          ? <CoursesList data={data} />
-          : <LinearProgress />
-        }
-      </div>
-    );
-  }
-}
+  const classes = useStyles();
+
+  return (
+    <div>
+      <Paper>
+        <Box p={2}>
+          <InputBase
+            fullWidth autoFocus
+            startAdornment={
+              <InputAdornment position="start">
+                <SearchIcon/>
+              </InputAdornment>
+            }
+            endAdornment={
+              isFiltering ? (
+                <InputAdornment position="end">
+                  <CircularProgress color="inherit"/>
+                </InputAdornment>
+              ) : null
+            }
+            placeholder="Start typing to filter by university name, course code, subject..." onChange={onQueryChange}/>
+        </Box>
+      </Paper>
+
+      <Box mt={2} mb={2}>
+        Matched {filteredCourses.length} out of {courses.total ? courses.total : 0} results
+      </Box>
 
-function mapStateToProps(state) {
-  return state
-}
+      <CoursesList data={filteredCourses}/>
 
-const mapDispatchToProps = {
-  getCourses
-}
+      <Backdrop open={courses.loading} className={classes.backdrop}>
```

Comment:
@Blu3spirits what do you think about using [skeletons](https://material-ui.com/api/skeleton/) here instead of the backdrop loader?

Reply:
Try it out and see. I like the idea of it but don't know how the implementation would look off the top of my head

Reply:
ae7a02b:

![ezgif com-video-to-gif](https://user-images.githubusercontent.com/7410405/80854593-f5cc9180-8bfe-11ea-894f-8acfa7736fda.gif)

(Search field is disabled until load is complete.)

I like it, although the skeletons do cause moderate CPU usage in both Firefox and Chrome.

Reply:
For desktop, this would be fine, but mobile users would have it rough. I think we should go back to how it was.

Tested this on my cell and confirmed it's not a very good experience and the old way was better IMO

Reply:
Reverted.